Framework/MDAlgorithms/src/GetSpiceDataRawCountsFromMD.cpp

```diff
@@ -499,17 +502,11 @@ MatrixWorkspace_sptr GetSpiceDataRawCountsFromMD::createOutputWorkspace(
     throw std::runtime_error("Failed to create output matrix workspace.");
 
   // Set data
-  MantidVec &dataX = outws->dataX(0);
-  MantidVec &dataY = outws->dataY(0);
-  MantidVec &dataE = outws->dataE(0);
-  for (size_t i = 0; i < sizex; ++i) {
-    dataX[i] = vecX[i];
-    dataY[i] = vecY[i];
-    if (dataY[i] > 1.)
-      dataE[i] = sqrt(dataY[i]);
-    else
-      dataE[i] = 1.;
-  }
+  outws->setHistogram(0, Points(std::move(vecX)), Counts(std::move(vecY)));
+
+  auto &dataE = outws->mutableE(0);
+  std::transform(vecY.begin(), vecY.end(), dataE.begin(),
```

Comment:
Yes. `remove_if` that the algorithm I was thinking of. And you're right about it breaking the E data anyway with the move. I guess the E value isn't actually covered by a test...